src/custom.test.js

```diff
@@ -0,0 +1,263 @@
+import {
+  CustomExportExtendedPublicKey,
+  CustomSignMultisigTransaction,
+} from "./custom";
+import { MAINNET, TESTNET, TEST_FIXTURES } from "unchained-bitcoin";
+import { INFO, PENDING, ACTIVE, ERROR } from "./interaction";
+import { customFixtures } from "./custom.fixtures";
+
+const { multisigs, transactions } = TEST_FIXTURES;
+
+describe("CustomExportExtendedPublicKey", () => {
+  function interactionBuilder({ bip32Path, network }) {
+    return new CustomExportExtendedPublicKey({
+      bip32Path,
+      network,
+    });
+  }
+
+  describe("constructor", () => {
+    it("fails with invalid network", () => {
+      expect(() => interactionBuilder({ network: "foob" })).toThrow(
+        /Unknown network/i
+      );
+    });
+    it("invalid bip32Path unsupported", () => {
+      const interaction = interactionBuilder({
+        network: TESTNET,
+        bip32Path: "m/45'/1/01",
+      });
+      expect(interaction.isSupported()).toBe(false);
+      expect(
+        interaction.hasMessagesFor({
+          state: PENDING,
+          level: ERROR,
+          code: "custom.bip32_path.path_error",
+        })
+      ).toBe(true);
+    });
+  });
+
+  describe("parse", () => {
+    it("fails when sending in nothing or non json", () => {
+      const notJSON = "test";
+      const definitelyNotJSON = 77;
+      const interaction = interactionBuilder({
+        network: TESTNET,
+        bip32Path: "m/45'/1/0",
+      });
+      expect(() => interaction.parse(notJSON)).toThrow(
+        /Not a valid ExtendedPublicKey/i
+      );
+      expect(() => interaction.parse(definitelyNotJSON)).toThrow(
+        /Not a valid ExtendedPublicKey/i
+      );
+      expect(() => interaction.parse({})).toThrow(
+        /Not a valid ExtendedPublicKey/i
+      );
+    });
+
+    it("fails when missing xpub field", () => {
+      const interaction = interactionBuilder({
+        network: MAINNET,
+        bip32Path: "m/45'/1'/0'",
+      });
+      const missingXpub = {
+        ...customFixtures.validCustomTpubJSON,
+      };
+      Reflect.deleteProperty(missingXpub, "xpub");
+      expect(() => interaction.parse(missingXpub)).toThrow(
+        /Not a valid ExtendedPublicKey/i
+      );
+    });
+
+    it("testnet no rootFingerprint is included so it gets computed", () => {
```

Comment:
some of the wording of the tests feel a little weird to me. I like to have it read as if it works in a sentence, e.g. the previous one `it fails when missing xpub field`. Not all of them are like that though. This is partially just personal preference, but I think it would be good to be consistent.

Reply:
Word, good point re: consistency - edited some of the names.